lib/app/utils/validation.rb

```diff
@@ -133,7 +142,23 @@ def self.guess_r4_profile(resource)
       return if resource.blank?
 
       candidates = RESOURCES[:r4][resource.resourceType]
-      return candidates.first if candidates.present?
+      return if candidates.blank?
+
+      if resource.resourceType == 'Observation'
+        return DEFINITIONS[US_CORE_R4_URIS[:smoking_status]] if resource&.code&.coding&.any? { |coding| coding&.code == '72166-2' }
+
+        return DEFINITIONS[US_CORE_R4_URIS[:observation_results]] if resource&.category&.coding&.any? { |coding| coding&.code == 'laboratory' }
+
+        return DEFINITIONS[US_CORE_R4_URIS[:pediatric_bmi_age]] if resource&.code&.coding&.any? { |coding| coding&.code == '59576-9' }
+
+        return DEFINITIONS[US_CORE_R4_URIS[:pediatric_weight_height]] if resource&.code&.coding&.any? { |coding| coding&.code == '77606-2' }
+      elsif resource.resourceType == 'DiagnosticReport'
+        return DEFINITIONS[US_CORE_R4_URIS[:diagnostic_report_lab]] if resource&.category&.coding&.any? { |coding| coding&.code == 'LAB' }
+
+        return DEFINITIONS[US_CORE_R4_URIS[:diagnostic_report_note]]
```

Comment:
Doesn't this still need a condition?

Reply:
So I'm still not sure about this. There isn't an easy restriction for diagnostic report note like there is for diagnostic report lab. So in the end, I just decided that if every diagnostic report is either a lab or a note for our testing purposes, then this approach would work. If the diagnostic report isn't a note or lab, then our tests won't be able to handle that anyway. But we should definitely discuss if there's a better way to handle this.

Reply:
Yeah, there doesn't seem to be a different way to differentiate them.